crates/revm/src/evm.rs

```diff
@@ -346,8 +346,8 @@ impl<EXT, DB: Database> Host for Evm<'_, EXT, DB> {
     fn log(&mut self, address: Address, topics: Vec<B256>, data: Bytes) {
         self.context.evm.journaled_state.log(Log {
             address,
-            topics,
-            data,
+            data: LogData::new(topics, data)
+                .expect("Invalid LogData: Number of topics should be <= 4"),
```

Comment:
Would refactor `Host::log` function input to a `LogData `.

in essence change this: https://github.com/bluealloy/revm/blob/main/crates/interpreter/src/instructions/host.rs#L214

Reply:
yea that's nicer

Reply:
I didn't notice as the code was not changed but we could use `Log` directly in place of `LogData`. Other parts lgtm

Reply:
could you clarify that last comment? not clear on what you're meaning